network-config-analyzer/TcpLikeProperties.py

```diff
@@ -120,155 +124,66 @@ def get_properties_obj(self):
-
+        assert not self.excluded_named_ports
+        assert not isinstance(other, TcpLikeProperties) or not other.named_ports
+        assert not isinstance(other, TcpLikeProperties) or not other.excluded_named_ports
+        super().__iand__(other)
         return self
 
     def __ior__(self, other):
-        self.cubes_set |= other.cubes_set
-
-        res_named_ports = dict({})
-        for port_name in self.named_ports:
-            res_named_ports[port_name] = self.named_ports[port_name]
-        for port_name in other.named_ports:
-            if port_name in res_named_ports:
-                res_named_ports[port_name] |= other.named_ports[port_name]
-            else:
-                res_named_ports[port_name] = other.named_ports[port_name]
-        self.named_ports = res_named_ports
-
-        res_excluded_named_ports = dict({})
-        for port_name in self.excluded_named_ports:
-            res_excluded_named_ports[port_name] = self.excluded_named_ports[port_name]
-        for port_name in other.named_ports:
-            if port_name in res_excluded_named_ports:
-                res_excluded_named_ports[port_name] -= other.named_ports[port_name]
-        self.excluded_named_ports = res_excluded_named_ports
-
+        assert not isinstance(other, TcpLikeProperties) or not other.excluded_named_ports
+        super().__ior__(other)
+        if isinstance(other, TcpLikeProperties):
+            res_named_ports = dict({})
+            for port_name in self.named_ports:
+                res_named_ports[port_name] = self.named_ports[port_name]
+            for port_name in other.named_ports:
+                if port_name in res_named_ports:
+                    res_named_ports[port_name] |= other.named_ports[port_name]
+                else:
+                    res_named_ports[port_name] = other.named_ports[port_name]
+            self.named_ports = res_named_ports
         return self
 
     def __isub__(self, other):
-        self.cubes_set -= other.cubes_set
-
-        res_named_ports = dict({})
-        for port_name in self.named_ports:
-            res_named_ports[port_name] = self.named_ports[port_name]
-        for port_name in other.named_ports:
-            if port_name in res_named_ports:
-                res_named_ports[port_name] -= other.named_ports[port_name]
-        self.named_ports = res_named_ports
-
-        res_excluded_named_ports = dict({})
-        for port_name in self.excluded_named_ports:
-            res_excluded_named_ports[port_name] = self.excluded_named_ports[port_name]
-        for port_name in other.named_ports:
-            if port_name in res_excluded_named_ports:
-                res_excluded_named_ports[port_name] |= other.named_ports[port_name]
-            else:
-                res_excluded_named_ports[port_name] = other.named_ports[port_name]
-        self.excluded_named_ports = res_excluded_named_ports
-
+        assert not self.excluded_named_ports
```

Comment:
can replace the 3 assertions with the following:
```
assert not self.has_named_ports()
assert not isinstance(other, TcpLikeProperties) or not other.has_named_ports()
```

Reply:
Done